runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -11970,9 +11970,9 @@ J9::Z::TreeEvaluator::VMinlineCompareAndSwap(TR::Node *node, TR::CodeGenerator *
    // compression sequence.
    bool isValueCompressedReference = false;
 
-   TR::Node* decompressedValueNode = newVNode;
+   TR::Node* compressedValueNode = newVNode;
```

Comment:
Thanks for renaming this !!

Reply:
I ended up getting the names backwards. That node is actually the uncompressed value. It was `compressedValueRegister` that I really wanted to rename since that doesn't contain a compressed value. I'll fix this.

Reply:
You are right, it does take the compressedRef object and then traverses through, to get the decompressed reference object, though it looked bit confusing, retracting my original comment, the name makes sense.

Reply:
I looked over the code again.

I am changing `compressedValueNode` back to `decompressedValueNode`. For objects under compressedrefs, this will be the node that represents the uncompressed object address found by searching down the tree. If compressedrefs are disable or the shift value is 0, it will be set to `newVNode` since the compressed and uncompressed object addresses are the same. For non-objects (byte, short, int, long), the naming will be a bit strange since non-addresses don't get compressed. But, I think it still conveys the meaning that this isn't a compressed value even if it is for values that never get compressed.

I will change `compressedValueRegister` to `decompressedValueRegister`. `compressedValueRegister` never holds a compressed object address so the name is clearly wrong. It comes from evaluating `decompressedValueNode` which is either an uncompressed object address or a non-object (byte, short, int, long).

How does that sound?

Reply:
Sounds good, thanks a lot Jimmy..

Reply:
Sounds good.  Thanks!